third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -319,6 +343,31 @@ pub struct FieldHandle {
     pub field: MemberCount,
 }
 
+/// A variant field access info
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+#[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
+pub struct VariantFieldHandle {
+    /// The structure which defines the variant.
+    pub owner: StructDefinitionIndex,
+    /// The sequence of variants which share the field at the given
+    /// field offset.
+    pub variants: Vec<VariantCount>,
```

Comment:
This `variants` field is redundant with the `VariantDefinition` field `fields`, as it should be the inverse.  If you keep both, you need a check for that on all input paths.  (Maybe you do have, this PR is too big to tell).  OTOH, if you get rid of this one, you can reconstruct it at deserialization and reduce the file size a bit.

Reply:
@wrwg So this means that we can have multiple VariantFieldHandle with the same owner? Just confirming